### PR TITLE
HTML: Use paragraphs instead of double line breaks

### PR DESCRIPTION
**Pull request by annando**
We now assume that double line feeds imply a new paragraph. This is more in line with the behaviour of other systems. Also it ensures more robustness, since HTML errors like missing ending elements will be encapsulated inside of paragraphs. This will be fixed by the HTML purifier.

### CODE REVIEW

**Review by MrPetovan — APPROVED** [review]

**Comment by nupplaphil** [comment]
Can we add a test for this? Changing something in `BBCode` is normally easily testable :-)

**Comment by annando** [comment]
@nupplaphil sure. But only later this evening.

**Review by MrPetovan — APPROVED** [review]
This is both smart and horrifying at the same time. HTMLPurifier is becoming load-bearing for our BBCode implementation 😅

**Comment by annando** [comment]
Yeah, I implemented some code cleanup:
```
$doc = new DOMDocument();
@$doc->loadHTML(mb_convert_encoding($text, 'HTML-ENTITIES', 'UTF-8'));
$node = $doc->getElementsByTagName('body')->item(0); 
$text = implode(array_map([$node->ownerDocument, 'saveHTML'], iterator_to_array($node->childNodes)));
```
Then I realized that the purifier is doing a great job at it already. I tested it very intensively yesterday.

File: tests/src/Content/Text/BBCodeTest.php [hunk 1]
```diff
@@ -259,7 +259,7 @@ public function dataBBCodes()
 				'text' => '[emoji=https://fedi.underscore.world/emoji/custom/custom/heart_nb.png]:heart_nb:[/emoji]',
 			],
 			'task-12900-multiple-paragraphs' => [
-				'expectedHTML' => '<h1>Header</h1><ul class="listbullet" style="list-style-type:circle;"><li>One</li><li>Two</li></ul><p>This is a paragraph<br>with a line feed.</p><p>Second Chapter</p>',
+				'expectedHTML' => "<h1>Header</h1><ul class=\"listbullet\" style=\"list-style-type:circle;\"><li>One</li><li>Two</li></ul><p>This is a paragraph<br>with a line feed.</p><p>Second Chapter</p>",
```
Comment (MrPetovan) [inline tests/src/Content/Text/BBCodeTest.php]: Why the change here?

Comment (annando) [reply tests/src/Content/Text/BBCodeTest.php]: Because there is some `\n` inside.

Comment (annando) [reply tests/src/Content/Text/BBCodeTest.php]: And I'm of course an idiot, since I did it with the wrong string :grin:

**Comment by MrPetovan** [comment]
> Tests should now work

I have bad news.

**Comment by annando** [comment]
The test now fails here:
```
There was 1 failure:

1) Friendica\Test\src\Core\KeyValueStorage\DBKeyValueStorageTest::testUpdatedAt with data set "boolTrue" ('data', true)
Failed asserting that 1679076668 is equal to 1679076667 or is less than 1679076667.

/woodpecker/src/github.com/friendica/friendica/tests/src/Core/KeyValueStorage/DBKeyValueStorageTest.php:83
```
And I really can't imagine that his can be influenced in any way via the BBCode parser.

@nupplaphil to the rescue!

**Comment by MrPetovan** [comment]
It's a timing issue that should be solved using the Clock object but it's outside the scope of this PR.
